chain/network/src/network_protocol/tests.rs

```diff
@@ -170,3 +176,96 @@ fn serialize_deserialize() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn serialize_deserialize_ip_addr() {
+    let mut rng = make_rng(89028037453);
+
+    // Convert IpAddr to proto::IpAddr for protocol buffer serialization
+    let ip_addr_before_serialize: net::IpAddr = data::make_ipv4(&mut rng);
+    let proto_ip_addr_before_serialize: proto::IpAddr = proto::IpAddr::from(&ip_addr_before_serialize);
+    // Serialize and deserialize for transporting over the network
+    let ip_addr_bytes: Vec<u8> = proto_ip_addr_before_serialize.write_to_bytes().unwrap();
+    let proto_ip_addr_after_deserialize = proto::IpAddr::parse_from_bytes(&ip_addr_bytes).unwrap();
+    assert_eq!(proto_ip_addr_before_serialize,  proto_ip_addr_after_deserialize);
+    // Convert proto::IpAddr back to IpAddr to be independent of protocol buffer in near network
+    let result_deserialized_actual_ip_addr = net::IpAddr::try_from(&proto_ip_addr_after_deserialize);
+    assert!(result_deserialized_actual_ip_addr.is_ok());
+    let deserialized_actual_ip_addr = result_deserialized_actual_ip_addr.unwrap();
+    assert_eq!(ip_addr_before_serialize, deserialized_actual_ip_addr);
+}
+
+#[test]
+fn sign_and_verify_ip_addr_with_peer_id() {
+    // Design sign and verify algorithm for std::net::IpAddr
+    let seed = "123";
+    let node_key = SecretKey::from_seed(KeyType::ED25519, seed);
+    let peer_id = PeerId::new(node_key.public_key());
+    let mut rng = make_rng(89028037453);
+    let ip_addr: net::IpAddr = data::make_ipv4(&mut rng);
+    let ip_bytes: Vec<u8> = match ip_addr {
+        net::IpAddr::V4(ip) => ip.octets().to_vec(),
+        net::IpAddr::V6(ip) => ip.octets().to_vec(),
+    };
+    let signature = node_key.sign(&ip_bytes);
+    assert!(signature.verify(&ip_bytes, &node_key.public_key()));
```

Comment:
This seems like a test of the `sign` function (which accepts an arbitrary sequence of bytes), which should have its own unit tests.

Reply:
The sign function is independent of ip data, this test is specific to checking the sign function works with our serialized approach for ip data that we discussed.

Reply:
What does this test catch? The fact that `.octets().to_vec()` produces a `Vec<u8>` is something checked by the compiler. The fact that `sign` will work on an arbitrary sequence of bytes is a test of the `sign` function. I don't think this test justifies its existence here.

Reply:
this shows how the interface of the new SignedIpAddress object looks like from the client code perspective. Regardless, remove as requested.